File: 11.盛最多水的容器.cs
using System;
/*
 * @lc app=leetcode.cn id=11 lang=csharp
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
public class Solution
{
    public int MaxArea(int[] height)
    {
        int res = 0;
        int left = 0;
        int right = height.Length - 1;
        while (left < right)
        {
            int temp = (right - left) * Math.Min(height[left], height[right]);
            res = temp > res ? temp : res;
            if (height[left] <= height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return res;
    }
}
// @lc code=end

File: 11.盛最多水的容器.rs
/*
 * @lc app=leetcode.cn id=11 lang=rust
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut left: usize = 0;
        let mut right = height.len() - 1;
        while left < right {
            let temp = (right - left) as i32 * Ord::min(height[left], height[right]);
            if temp > res {
                res = temp;
            }
            if height[left] <= height[right] {
                left += 1;
            } else {
                right -= 1;
            }
        }
        res
    }
}
// @lc code=end
